### PR TITLE
DXCDT-377: E2E test cases for keyword presevation

### PR DESCRIPTION
**Pull request by willvedd**
### 🔧 Changes

While many unit tests were added alongside the integration of the `preserveKeywords` function into the export process (#751), there is still a need for E2E test cases that simulate more real scenarios. This PR adds these test cases for fuller testing coverage of the keyword preservation feature.

The work done in this PR has highlighted two key pieces of functionality missing:
- Preserving keywords in auxillary files (ex: HTML templates, JS files)
- Preservation of `@@` array replacement syntax in YAML files due to invalid syntax if left as-is and invalid if wrapped in quotes.
Both of the above have test cases written and commented-out but will be addressed in the coming days.

### 📚 References

Keyword Preservation RFC: #688 
Related implementation PR: #751 

### 🔬 Testing

This PR only adds testing.

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: test/e2e/e2e.test.ts [hunk 1]
```diff
@@ -358,3 +358,84 @@ describe('#end-to-end keyword replacement', function () {
...
+    const yaml = yamlLoad(fs.readFileSync(path.join(workDirectory, 'tenant.yaml')));
+    expect(yaml.tenant.friendly_name).to.equal('##TENANT_NAME##');
+    expect(yaml.tenant.support_email).to.equal('support@##DOMAIN##');
+    expect(yaml.tenant.support_url).to.equal('https://##DOMAIN##/support');
+    // expect(yaml.tenant.enabled_locales).to.equal('@@LANGUAGES@@'); TODO: enable @@ARRAY@@ keyword preservation in yaml formats
+
+    // const emailTemplateHTML = fs
+    //   .readFileSync(path.join(workDirectory, 'emailTemplates', 'welcome_email.html'))
+    //   .toString();
+    // expect(emailTemplateHTML).to.contain('##TENANT##'); TODO: enable keyword preservation in auxillary template files
```
Comment (willvedd) [inline test/e2e/e2e.test.ts:404]: This and the similar test below will be added in the coming days when the relevant fix has been added.

File: test/e2e/testdata/should-preserve-keywords/yaml/tenant.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+tenant:
+  allowed_logout_urls:
+    - https://travel0.com/logoutCallback
+  # enabled_locales: @@LANGUAGES@@ TODO: enable @@ARRAY@@ keyword preservation in yaml
```
Comment (willvedd) [inline test/e2e/testdata/should-preserve-keywords/yaml/tenant.yaml:4]: This also requires a fix that will come later.

File: test/e2e/testdata/should-preserve-keywords/yaml/tenant.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      page_background: '#222221'
+emailTemplates:
+  - template: welcome_email
+    body: ./emailTemplates/welcome_email.html
+    enabled: false
+    from: ''
+    resultUrl: https://##DOMAIN##/welcome
+    subject: Welcome
+    syntax: liquid
+    urlLifetimeInSeconds: 3600
```
Comment (sergiught) [inline test/e2e/testdata/should-preserve-keywords/yaml/tenant.yaml:42]:
```suggestion
    urlLifetimeInSeconds: 3600

```

File: test/e2e/recordings/should-preserve-keywords-for-directory-format.json [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+            "from": "",
+            "subject": "",
+            "syntax": "liquid",
+            "urlLifetimeInSeconds": 432000,
+            "enabled": true
+        },
+        "rawHeaders": [],
+        "responseIsBinary": false
+    }
+]
```
Comment (sergiught) [inline test/e2e/recordings/should-preserve-keywords-for-directory-format.json:426]:
```suggestion
]

```

File: test/e2e/recordings/should-preserve-keywords-for-yaml-format.json [hunk 5]
```diff
@@ -0,0 +1,450 @@
...
+        "response": {
+            "statusCode": 404,
+            "error": "Not Found",
+            "message": "The template does not exist.",
+            "errorCode": "inexistent_email_template"
+        },
+        "rawHeaders": [],
+        "responseIsBinary": false
+    }
+]
```
Comment (sergiught) [inline test/e2e/recordings/should-preserve-keywords-for-yaml-format.json:450]:
```suggestion
]

```

**Review by sergiught — APPROVED** [review]

**Pull request merged**